tool/public/client.js

```diff
@@ -9,28 +9,45 @@ class ConverterUi {
 
 		iptFile.addEventListener("change", () => {
 			const reader = new FileReader();
-			reader.onload = () => iptText.value = reader.result;
 			reader.readAsText(iptFile.files[0]);
-			doConvert();
+			reader.onload = () => {
+				let text;
+
+				// Check whether filetype is legal
+				if (iptFile.value.match(/\.(json|txt)$/i) || iptFile.value.match(/$(?<!\.\w+)/)) { // .json, .txt, or no filetype
+					text = reader.result;
+				} else if (iptFile.value.match(/\.db$/i)) { // .db
+					text = `[\n\t${reader.result.replace(/\}\n\{/g, "},\n\t{")}]`; // Hastily converting to JSON
+				} else {
+					outText.value = "Failed to parse input text!\n\n> Invalid filetype";
+					console.error(`Failed to load invalid filetype: ${iptFile.value.match(/(?<=\/|\\)[^\/\\]+$/)?.[0]}`); // This monstrosity just grabs the filename from the path
+					return;
+				}
+
+				iptText.value = text;
+				doConvert();
+			};
 		});
 
 		const doConvert = () => {
-			let json;
 			try {
-				json = JSON.parse(iptText.value);
+				outText.value = JSON.stringify(Converter.getConverted(JSON.parse(iptText.value)), null, "\t");
 			} catch (e) {
-				console.log("Failed to parse input text!");
+				outText.value = `Failed to parse input text!\n\n${e}`;
 				throw e;
 			}
-			outText.value = JSON.stringify(Converter.getConverted(json), null, "\t");
 		};
 
 		btnConvert.addEventListener("click", () => doConvert());
 
 		btnCopy.addEventListener("click", async () => {
 			await navigator.clipboard.writeText(outText.value);
+			btnCopy.innerHTML = "Copy ✓";
 			console.log("Copied!");
+			window.setTimeout(btnCopy.resetText, 480);
 		});
+
+		btnCopy.resetText = () => btnCopy.innerHTML = "Copy";
```

Comment:
this is needless scope escalation; inline it in the click handler as  `window.setTimeout(() => btnCopy.innerHTML = "Copy"), 480);` instead

Reply:
This was actually my first move, but for some reason it just didn't work and I spent like 20 minutes figuring out why not. It worked first-time once I named the function, so I stopped fighting and got on with other things 🥲

It now works, so I guess I had made an extremely stupid typo somewhere and never noticed